JOSESwift/Sources/JWS.swift

```diff
@@ -152,6 +154,44 @@ public struct JWS {
 
         return self
     }
+
+    /// Checks whether the JWS's signature is valid using a given verifier.
+    ///
+    /// - Parameter verifier: The verifier containing the public key whose corresponding private key signed the JWS.
+    /// - Returns: The JWS on which this function was called if the signature is valid.
+    /// - Throws: A `JOSESwiftError` if the signature is invalid or if errors occured during signature validation.
+    public func validate(using verifier: Verifier) throws -> JWS {
+        guard verifier.verifier.algorithm == header.algorithm else {
```

Comment:
This looks really strange imo. Could we maybe rename the "verifierProtocol" to something else? Can't come up with a better name though.

Reply:
I currently can only think of `Verifier` and `Validator` but that will be strange too, having two different names (although I do like validator more than verifier somehow 😅). We could rename the parameter in the validate method. But to what? 😅